### ISSUE TITLE
Operator: output syntax unfit + error

### ISSUE THREAD

**Issue by flexoron** [infradig/trealla#396]
```
Trealla Prolog (c) Infradig 2020-2021, v1.19.27

?- op(200,xf,oP).
   true.

?- E = 1 oP.
   E = 1oP. % unexpected, expecting E = 1 oP.

?- E = 1oP.
Error: syntax error, parsing number, line 1, ''

% shouldn't be an error because operand is numeric
```

**Comment by UWN** [comment]
This is a bit an ambiguity. Think of [#256](http://www.complang.tuwien.ac.at/ulrich/iso-prolog/conformity_testing#256). [7.10.5 h 1](www.complang.tuwien.ac.at/ulrich/iso-prolog/draft-7.10.5#h_1):

> ... In all cases, a space is output
to separate an operator from its argument(s) if any
ambiguity could otherwise arise.

So what does *any ambiguity* mean? Syntax wise, there is no ambiguity whatsoever, still making a space could make sense here.

There are, however, uses of such postfix operator for (SI) units.  Think of `X uis 1m+10mm,`. Is `X = 1010 mm` OK as an answer, or isn't rather `X = 1010mm` preferred?

What is certainly incorrect is that `1oP` is not parsed as `oP(1)`.

**Comment by UWN** [comment]
Cases of bigger ambiguity are those that start with `0`, since `0o` starts an octal number, still `0oP` would be valid syntax since we have an *eager consumer rule* for tokenwise parsing. More uglitudes of that kind are [#251 ff](http://www.complang.tuwien.ac.at/ulrich/iso-prolog/conformity_testing#251).

**Comment by flexoron** [comment]
```
X is 1m+1mm. X = 1.001m
X is 1mm+1m. X = 1001mm
```

**Comment by flexoron** [comment]
v1.20.11
```
?- op(200,xf,oP).

?- E = aoP.
   E = aoP.  % expected

?- E = a oP.
   E = aoP.  % unexpected, should be E = a oP, see below

% bin,oct,hex should be catched
?- 0b1oP.
Error: syntax error, parsing binary, line 1, ' % unexpected, see below
?- 
```
scryer,gnu,swi
```
?- op(200,xf,oP).
   true.

?- E = a oP.
   E = a oP.

?- 0b1oP.
caught: error(existence_error(procedure,(oP)/1),(oP)/1) % expected

?- E = 0b1oP.
   E = 1 oP.
```

**Comment by flexoron** [comment]
Many thanks!...
...fixed with v1.20.12
